week2/internal/repository/implement/actor_repository.go

```diff
@@ -41,3 +41,38 @@ func (repo *ActorRepository) CreateActor(ctx context.Context, actor *entity.Acto
 	}
 	return nil
 }
+
+func (repo *ActorRepository) UpdateActor(ctx context.Context, actor *entity.Actor) error {
+	//update all columns of an actor
+	if err := repo.db.WithContext(ctx).Where("actor_id = ?", actor.ID).Save(actor).Error; err != nil {
```

Comment:
Why using ```Save``` here. I've search through the document and don't see any usage of ```Where``` combining with ```Save```. Also refer to this documentation
``` Save is a combination function. If save value does not contain primary key, it will execute Create, otherwise it will execute Update (with all fields).```
So please use ```Updates``` instead to prevent Creation of non exists record

Reply:
Maybe you can try using only ```Save```. Remove the ```Where``` statement, also update the tag of ```Actor``` entity. Make it primaryKey so the ```Save``` operation won't create new record if ```ID``` not exists

Reply:
Please refer to this documentation: https://gorm.io/docs/update.html

Reply:
> NOTE Don’t use Save with Model, it’s an Undefined Behavior.

Read through the document to see if you are doing it right

Reply:
I use `Where` to check if actor is exist. Although it don't have any document about this but it helps code clearly right?  Beside i have added `actor_id` field into Actor before using `Save` so we always have primary key here.

Reply:
Please read the document and remove the ```Where``` statement. I don't see any usage of ```Where``` combining with ```Save```. Also the ```Save``` document already declare that:

> Save is a combination function. If save value does not contain primary key, it will execute Create, otherwise it will execute Update (with all fields).